cmd/catchupsrv/main.go

```diff
@@ -32,6 +34,7 @@ import (
 
 var addrFlag = flag.String("addr", "127.0.0.1:4160", "Address to listen on")
 var dirFlag = flag.String("dir", "", "Directory containing catchup blocks")
+var subfoldersFlag = flag.Bool("subfolders", false, "Organize downloaded blocks in subfolders (if used, must be specified both when executed with and without -download)")
```

Comment:
Should this always be true (i.e., not give the user an option)?  Is there a compelling reason to support the flat structure and the sub-directory structure as well?

Reply:
Flat directory structure tend to become *very* slow once number of files exceed certain count.
( I don't remember the precise metrics ). 
I think that supporting a flat directory structure is nice ( in theory, that it.. ), but not practical for long-term use.

To be more precise, when you have many files in a single directory - 
- traversal on the files list becomes very slow ( i.e. without opening the actual files )
- adding/removing a file can take a long time.

The above two goes away when you have the files in structured directory.

Reply:
@rotemh , fabrice says you were opposed to unilaterally turning on the subfolder structure.  Is there an important use case of keeping all these files in a single directory?

Reply:
Just saw this -- no objection, just said then that we need to make a general solution and nothing ad-hoc.

Reply:
Thanks.  I think then we should make `catchupsrv` always use the nested directory structure, for simplicity.